### PR TITLE
store monitoring updates

### PR DESCRIPTION
**Pull request by Yibaebi**

### CODE REVIEW

File: app/styles/_color.scss [hunk 1]
```diff
@@ -84,6 +84,7 @@ body {
   --font-weight-medium: 600;
   --font-weight-bold: 700;
 
+  --zindex-notification: 1099;
```
Comment (future-pirate-king) [inline app/styles/_color.scss:87]: why is this required ?

File: app/templates/authenticated/app-monitoring.hbs [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {{page-title 'Store Monitoring'}}
 
-<AppMonitoring @appMonitoring={{@model}} />
```
Comment (future-pirate-king) [inline app/templates/authenticated/app-monitoring.hbs:3]: is this file required ?

File: tests/integration/components/app-monitoring/details-table/store-version-test.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupIntl(hooks);
+    setupMirage(hooks);
+
+    hooks.beforeEach(async function () {
+      this.store = this.owner.lookup('service:store');
+
+      // Add file, am-app, am-app-sync, am-app-version, and project records to store
+      this.store.push({
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/details-table/store-version-test.js:21]: you can use mirage factories here

File: tests/integration/components/app-monitoring/details-table/store-version-test.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      this.server.get('/v2/am_app_versions/:id', () => {
+        return {
+          id: '1',
+        };
+      });
+    });
+
+    test('it renders correctly when amApp is "SCANNED"', async function (assert) {
+      // The amApp record pushed to the store below represents a "SCANNED" state.
+      this.store.push({
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/details-table/store-version-test.js:112]: same mirage factories here

File: tests/integration/components/app-monitoring/details-table/store-version-test.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+      assert
+        .dom('[data-test-am-details-table-store-version-fileID]')
+        .exists()
+        .containsText('t:fileID:()')
+        .containsText(this.amAppVersion.get('latestFile').get('id'));
+    });
+
+    test('it renders correctly when amApp is "NOT SCANNED"', async function (assert) {
+      // The amApp record pushed to the store below represents a "NOT SCANNED" state.
+      this.store.push({
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/details-table/store-version-test.js:166]: same mirage factories here

File: tests/integration/components/app-monitoring/details-test.js [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+import styles from 'irene/components/app-monitoring/details/index.scss';
+import { module, test } from 'qunit';
+import faker from 'faker';
+
+module('Integration | Component | app-monitoring/details', function (hooks) {
+  setupRenderingTest(hooks);
+
+  setupIntl(hooks);
+
+  hooks.beforeEach(async function () {
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/details-test.js:15]: should use mirage, store models are ok if passed as argument

File: tests/integration/components/app-monitoring/history-table/scanned-status-test.js [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+module(
+  'Integration | Component | app-monitoring/history-table/scanned-status',
+
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupIntl(hooks);
+    setupMirage(hooks);
+
+    hooks.beforeEach(async function () {
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/history-table/scanned-status-test.js:17]: same mirage and factories should be used

File: tests/integration/components/app-monitoring/history-table/store-version-test.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupIntl(hooks);
+    setupMirage(hooks);
+
+    hooks.beforeEach(async function () {
+      this.store = this.owner.lookup('service:store');
+
+      // Add  am-app, am-app-sync, am-app-version, and project records to store
+      this.store.push({
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/history-table/store-version-test.js:21]: same mirage and factories should be used here

File: tests/integration/components/app-monitoring/table/store-version-test.js [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+import faker from 'faker';
+
+module(
+  'Integration | Component | app-monitoring/table/store-version',
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupMirage(hooks);
+    setupIntl(hooks);
+
+    hooks.beforeEach(async function () {
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/table/store-version-test.js:16]: same mirage and factories should be used

File: tests/integration/helpers/am-status-condition-test.js [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import faker from 'faker';
+
+module('Integration | Helper | am-status-condition', function (hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
```
Comment (future-pirate-king) [inline tests/integration/helpers/am-status-condition-test.js:12]: factories and store could be used here

File: tests/integration/helpers/am-status-text-test.js [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+module('Integration | Helper | am-status-text', function (hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+  setupIntl(hooks);
+
+  hooks.beforeEach(async function () {
+    this.store = this.owner.lookup('service:store');
+
+    this.store.push({
```
Comment (future-pirate-king) [inline tests/integration/helpers/am-status-text-test.js:17]: factories and store could be used here

File: app/components/ak-link/index.hbs [hunk 12]
```diff
@@ -28,7 +28,9 @@
     @noWrap={{@noWrap}}
     @tag='span'
     class={{@linkTextClass}}
-    local-class='ak-link-text'
+    local-class='
+    ak-link-text  
+    ak-link-text-font-weight-{{or @fontWeight "regular"}}'
```
Comment (future-pirate-king) [inline app/components/ak-link/index.hbs:33]: I think this should be at the level of `ak-typography`

File: app/controllers/authenticated/app-monitoring/monitoring-details.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import Controller from '@ember/controller';
+import AmAppModel from 'irene/models/am-app';
+
+export default class AuthenticatedAppMonitoringMonitoringDetailsController extends Controller {
+  declare model: AmAppModel;
```
Comment (future-pirate-king) [inline app/controllers/authenticated/app-monitoring/monitoring-details.ts:5]: whats the purpose of these controllers ?

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/ak-table/index.scss [hunk 14]
```diff
@@ -19,21 +19,21 @@
 
 .ak-table-padding-dense {
   table thead th {
-    padding: 0.5em 0.75em;
+    padding: 0.5em 1.5714em;
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/ak-table/index.scss:22]: The horizontal padding for the dense and default table variants were wrongly set. The cell contents become too close to the borders when borders are visible. The units are from the store monitoring table.

File: app/components/app-monitoring/details-table/country-code/index.hbs [hunk 15]
```diff
@@ -0,0 +1,3 @@
+<AkTypography data-test-details-table-country-code>
+  UK
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/app-monitoring/details-table/country-code/index.hbs:2]: Temporarily. This would be changed once the integration is done.

File: app/components/app-monitoring/details-table/index.hbs [hunk 16]
```diff
@@ -0,0 +1,28 @@
+<div class='px-3'>
+  {{#if this.hasNoAmAppVersions}}
+    <AppMonitoring::Empty
+      @header='{{t "notFound"}}'
+      @body='Lorem ipsum dolor sit amet consectetur. Maecenas quis morbi libero lacus. Cras sapien cursus bibendum.'
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/app-monitoring/details-table/index.hbs:5]: Yeah. The content would be updated once Raghu provides the right content. This would be updated before the release.

File: app/components/app-monitoring/details-table/store-version/index.hbs [hunk 17]
```diff
@@ -0,0 +1,39 @@
+{{! @glint-nocheck:  nested latestAmAppVersion properties}}
+<AkStack
+  @direction='column'
+  @justifyContent='center'
+  data-test-am-details-table-store-version
+>
+  {{#if this.amApp.latestAmAppVersion.comparableVersion}}
+
+    {{this.amApp.latestAmAppVersion.comparableVersion}}
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/app-monitoring/details-table/store-version/index.hbs:9]: What error?

File: app/controllers/authenticated/app-monitoring/monitoring-details.ts (hunk 13 above)
Comment (Yibaebi) [reply app/controllers/authenticated/app-monitoring/monitoring-details.ts:5]: Originally this was added for typing purposes at the template files. It is no longer required. They would be deleted.

File: app/styles/_color.scss (hunk 1 above)
Comment (Yibaebi) [reply app/styles/_color.scss:87]: It's from the previous commits. Once squashed, it should be fixed.

File: app/components/ak-link/index.hbs [hunk 18]
```diff
@@ -27,6 +27,7 @@
     @color={{if @disabled 'inherit' @color}}
     @noWrap={{@noWrap}}
     @tag='span'
+    @fontWeight={{or @fontWeight 'regular'}}
```
Comment (future-pirate-king) [inline app/components/ak-link/index.hbs:30]: there should be no default value here will affect `@typographyVariant `

File: app/components/app-monitoring/details-table/store-version/index.hbs (hunk 17 above)
[the remark this answers is not in the document]
Comment (future-pirate-king) [reply app/components/app-monitoring/details-table/store-version/index.hbs:9]: linting related error

File: app/components/ak-svg/am-loading.hbs [hunk 19]
```diff
@@ -0,0 +1,167 @@
+<svg
```
Comment (future-pirate-king) [inline app/components/ak-svg/am-loading.hbs:1]: This svg is not specific to app monitoring so can we rename this to something like `empty-loading-list.hbs`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — APPROVED** [review]
